llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -2041,6 +2041,28 @@ void CombinerHelper::applyCombineMulToShl(MachineInstr &MI,
   Observer.changedInstr(MI);
 }
 
+bool CombinerHelper::matchCombineSubToAdd(MachineInstr &MI,
+                                          BuildFnTy &MatchInfo) {
+  assert(MI.getOpcode() == TargetOpcode::G_SUB && "Expected a G_SUB");
```

Comment:
```
GSub *Sub = cast<GSub>(&MI);
```

Reply:
Up to you. I want to make your live easier.

Reply:
Yes, this should be cleaned up at some point